### DISCUSSION TITLE
Old version of jspsych on cognition.run (v5.0.3)

### DISCUSSION THREAD

**Discussion by hayleydorfman** [jspsych/jsPsych#1709] [Q&A] [answered]
Hello,
Has anyone had any luck running an old version of a jsPsych experiment on cognition.run? If so, how did you achieve this? I often recommend cognition.run to people who use my jsPsych experiments because it's so great, but I have one task made a few years ago in version 5.0.3 that some colleagues want to use, but they can't get it to work on cognition.run. I've tried to troubleshoot to see where I might make edits, but haven't had much luck. As far as I can understand, there is no way to stop cognition.run from using version 6+. I even tried uploading the 5.0.3 jspsych.js file in hopes it might "overwrite" whatever cognition.run was presetting but that didn't work. I would really like to avoid reprogramming the task in a newer version of jsPsych since the task is already published and validated. The task was previously run successfully on a personal lab server, but the group who wants to use the task does not have access to a server of their own, so using cognition.run would be ideal. Thanks in advance for any advice!

**Comment by becky-gilbert** [comment]
Hi @hayleydorfman, this might be something that @javidalpe can answer. But I'm surprised that you can't just upload the v5.0.3 jspsych.js file, as well as all of the plugins that you need. Do you have a live version that you could share?

**Reply by hayleydorfman** [reply]
Hi Becky, thanks for your help. I was surprised as well - I'm not sure what's going on. If I do upload the jspych.js file I get this error stemming from the .js file: "Uncaught ReferenceError: $ is not defined." If I remove the .js file then I get some console warnings but no errors, but the task won't run. Live version is here: [https://8pm55gaqde.cognition.run]

**Reply by becky-gilbert** [reply]
Ah ok, the "Uncaught ReferenceError: $ is not defined." error is happening because older versions of jsPsych required jquery, whereas the newer versions don't. So you can try loading the jquery file into cognition.run as well.

**Reply by hayleydorfman** [reply]
Hmm, I still get the same error after uploading a jquery-ui.css file from version 5.0.3. The uncaught reference error is directing me to this line in the jspsych.js file, if it helps:

// now add to list so that it gets appended to all future data
    dataProperties = $.extend({}, dataProperties, properties);
  };

**Reply by becky-gilbert** [reply]
Ok that's a CSS file, but I'm wondering if you see a jquery file with the JavaScript .js extension somewhere in your expeirment folder? I think the file name might be something like "jquery-3.6.0.min.js" but with a different (earlier) release number. I'm guessing you should also see that this file is loaded in a `script` tag in the `head` section of your HTML file.

**Comment by javidalpe** [comment, marked as answer]
Hello @hayleydorfman,

I'm sorry but cognition.run do not support jspsych 5.X at the moment. While you could make your experiment run on the platform (host the experiment and present it with no errors), there is no way cognition.run will understand where and how is the data stored in the client, therefore the data won't be sent to the server and it won't be available to download 😞 

My recommendation is to update jspsych  to 6.X. Is that something reasonable? @becky-gilbert  do you have a upgrade guide to help  people move from 5.X to 6.X and forward?

**Reply by hayleydorfman** [reply]
Thanks for the info, @javidalpe. I haven't been able to get it to run with Becky's jquery tip above, but I suppose it doesn't matter if the data won't save. I could upgrade the code to v6+ at some point, but it won't help my colleagues who are trying to use the experiment in the near future and don't have access to their own server. @becky-gilbert if there is an upgrade guide that would be incredibly helpful! Thank you both for your help, much appreciated!

**Reply by becky-gilbert** [reply]
That makes sense @javidalpe, thanks for letting us know!
@hayleydorfman unfortunately I'm not aware of a v5-v6 upgrade guide. All I can suggest is reading through the release notes (in particular the changes sections) and using the documentation for the version that you want to upgrade to as a guide. Feel free to post any questions and code in a separate discussion post and we'll do our best to help out!
